cpp/src/dbscan/dbscan_mg.cu

```diff
@@ -0,0 +1,109 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cuml/cuml_api.h>
+#include <common/cumlHandle.hpp>
+#include <cuml/cluster/dbscan_mg.hpp>
+#include "dbscan.cuh"
+
+namespace ML {
+namespace Dbscan {
+namespace opg {
+
+void fit(const raft::handle_t &handle, float *input, int n_rows, int n_cols,
+         float eps, int min_pts, int *labels, size_t max_bytes_per_batch,
+         int verbosity) {
+  dbscanFitImpl<float, int, true>(handle, input, n_rows, n_cols, eps, min_pts,
+                                  labels, nullptr, max_bytes_per_batch,
+                                  handle.get_stream(), verbosity);
+}
+
+void fit(const raft::handle_t &handle, double *input, int n_rows, int n_cols,
+         double eps, int min_pts, int *labels, size_t max_bytes_per_batch,
+         int verbosity) {
+  dbscanFitImpl<double, int, true>(handle, input, n_rows, n_cols, eps, min_pts,
+                                   labels, nullptr, max_bytes_per_batch,
+                                   handle.get_stream(), verbosity);
+}
+
+void fit(const raft::handle_t &handle, float *input, int n_rows, int n_cols,
+         float eps, int min_pts, int *labels, int *core_sample_indices,
+         size_t max_bytes_per_batch, int verbosity) {
+  dbscanFitImpl<float, int, true>(
+    handle, input, n_rows, n_cols, eps, min_pts, labels, core_sample_indices,
+    max_bytes_per_batch, handle.get_stream(), verbosity);
+}
+
+void fit(const raft::handle_t &handle, double *input, int n_rows, int n_cols,
+         double eps, int min_pts, int *labels, int *core_sample_indices,
+         size_t max_bytes_per_batch, int verbosity) {
+  dbscanFitImpl<double, int, true>(
+    handle, input, n_rows, n_cols, eps, min_pts, labels, core_sample_indices,
+    max_bytes_per_batch, handle.get_stream(), verbosity);
+}
+
+void fit(const raft::handle_t &handle, float *input, int64_t n_rows,
+         int64_t n_cols, float eps, int min_pts, int64_t *labels,
+         size_t max_bytes_per_batch, int verbosity) {
+  dbscanFitImpl<float, int64_t, true>(
+    handle, input, n_rows, n_cols, eps, min_pts, labels, nullptr,
+    max_bytes_per_batch, handle.get_stream(), verbosity);
+}
+
+void fit(const raft::handle_t &handle, double *input, int64_t n_rows,
+         int64_t n_cols, double eps, int min_pts, int64_t *labels,
+         size_t max_bytes_per_batch, int verbosity) {
+  dbscanFitImpl<double, int64_t, true>(
+    handle, input, n_rows, n_cols, eps, min_pts, labels, nullptr,
+    max_bytes_per_batch, handle.get_stream(), verbosity);
+}
+
+void fit(const raft::handle_t &handle, float *input, int64_t n_rows,
+         int64_t n_cols, float eps, int min_pts, int64_t *labels,
+         int64_t *core_sample_indices, size_t max_bytes_per_batch,
+         int verbosity) {
+  dbscanFitImpl<float, int64_t>(
```

Comment:
They do. The tests were passing despite my mistake because it falls back to the single-GPU implementation if the `opg` template parameter is not explicitly set to `true` (each worker computes the labels alone and doesn't communicate with the others, but it's correct).